tests/nested/manual/uc-update-assets-secure/task.yaml

```diff
@@ -0,0 +1,58 @@
+summary: Update boot assets through gadgets in secure boot
+
+systems: [ubuntu-20.04-64, ubuntu-22.04-64]
+
+environment:
+  NESTED_ENABLE_SECURE_BOOT: true
+
+prepare: |
+  CHANNEL="$(tests.nested show version)/stable"
+  snap download --basename=pc --channel="${CHANNEL}" pc
+  unsquashfs -d pc pc.snap
+  KEY_NAME=$(tests.nested download snakeoil-key)
+  SNAKEOIL_KEY="$PWD/$KEY_NAME.key"
+  SNAKEOIL_CERT="$PWD/$KEY_NAME.pem"
+
+  # Save the shim before resigning
+  cp pc/shim.efi.signed shim.efi.signed
+
+  # Repack pc gadget for the initial image
+  tests.nested secboot-sign file pc/shim.efi.signed "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  tests.nested secboot-sign file pc/grubx64.efi "$SNAKEOIL_KEY" "$SNAKEOIL_CERT"
+  snap pack pc "$(tests.nested get extra-snaps-path)"
+
+  # Remove signatures
+  cp shim.efi.signed shim.efi
+  while sbverify --list shim.efi | grep "^signature [0-9]*$"; do
+    sbattach --remove shim.efi
+  done
+
+  # Add a different vendor certificate
+  openssl req -new -x509 -newkey rsa:2048 -subj "/CN=new vendor certificate/" -keyout new-cert.key -out new-cert.crt -days 3650 -nodes -sha256
+  openssl x509 -outform der -in new-cert.crt -out new-cert
+  python3 generate_vender_cert_section.py new-section new-cert
+  objcopy --update-section .vendor_cert=new-section shim.efi shim.efi.out
+
+  # Sign modified shim
+  cp shim.efi.out pc/shim.efi.signed
+  tests.nested secboot-sign file pc/shim.efi.signed "${SNAKEOIL_KEY}" "${SNAKEOIL_CERT}"
+
+  # Resign grub with new vendor key
+  tests.nested secboot-sign file pc/grubx64.efi "new-cert.key" "new-cert.crt"
+
+  (cd pc && python3 "${TESTSLIB}/bump_asset_edition.py")
+
+  snap pack pc --filename=pc_2.snap
+
+  tests.nested build-image core
+  tests.nested create-vm core
+
+execute: |
+  tests.nested copy pc_2.snap
+  boot_id="$(tests.nested boot-id)"
+  tests.nested exec "sha256sum -b /boot/efi/EFI/boot/bootx64.efi" >old-hash
+  tests.nested exec "sudo snap install --dangerous pc_2.snap"
+  tests.nested wait-for reboot "${boot_id}"
+  tests.nested exec "sha256sum -b /boot/efi/EFI/boot/bootx64.efi" >new-hash
+  sed "s, .*, *pc/shim.efi.signed," new-hash | sha256sum -c
+  sed "s, .*, *pc/shim.efi.signed," old-hash | not sha256sum -c
```

Comment:
You would need also to check the grub update in ubuntu-seed here.

Reply:
I rewrote the checks. Now it checks all.